--- sapi/fpm/Makefile.frag.orig	2018-01-02 23:35:53.000000000 +0100
+++ sapi/fpm/Makefile.frag	2018-01-13 15:23:40.569321827 +0100
@@ -15,8 +15,8 @@
 	else \
 		echo "Installing PHP FPM defconfig:     $(INSTALL_ROOT)$(sysconfdir)/" && \
 		$(mkinstalldirs) $(INSTALL_ROOT)$(sysconfdir)/php-fpm.d; \
-		$(INSTALL_DATA) sapi/fpm/php-fpm.conf $(INSTALL_ROOT)$(sysconfdir)/php-fpm.conf.default; \
-		$(INSTALL_DATA) sapi/fpm/www.conf $(INSTALL_ROOT)$(sysconfdir)/php-fpm.d/www.conf.default; \
+		$(INSTALL_DATA) sapi/fpm/php-fpm.conf $(INSTALL_ROOT)$(sysconfdir)/php-fpm.conf; \
+		$(INSTALL_DATA) sapi/fpm/www.conf $(INSTALL_ROOT)$(sysconfdir)/php-fpm.d/www.conf; \
 	fi
 	
 	@echo "Installing PHP FPM man page:      $(INSTALL_ROOT)$(mandir)/man8/"
--- sapi/fpm/php-fpm.conf.in.orig	2018-01-02 23:35:53.000000000 +0100
+++ sapi/fpm/php-fpm.conf.in	2018-01-13 15:23:40.575988494 +0100
@@ -14,7 +14,7 @@
 ; Pid file
 ; Note: the default prefix is @EXPANDED_LOCALSTATEDIR@
 ; Default Value: none
-;pid = run/php-fpm.pid
+;pid = /run/php-fpm/php-fpm.pid
 
 ; Error log file
 ; If it's set to "syslog", log is sent to syslogd instead of being written
--- sapi/fpm/www.conf.in.orig	2018-01-02 23:35:53.000000000 +0100
+++ sapi/fpm/www.conf.in	2018-01-13 15:23:40.582655160 +0100
@@ -44,8 +44,8 @@
 ; BSD-derived systems allow connections regardless of permissions.
 ; Default Values: user and group are set as the running user
 ;                 mode is set to 0660
-;listen.owner = @php_fpm_user@
-;listen.group = @php_fpm_group@
+listen.owner = @php_fpm_user@
+listen.group = @php_fpm_group@
 ;listen.mode = 0660
 ; When POSIX Access Control Lists are supported you can set them using
 ; these options, value is a comma separated list of user/group names.
@@ -356,7 +356,7 @@
 ; Chdir to this directory at the start.
 ; Note: relative path can be used.
 ; Default Value: current directory or / when chroot
-;chdir = /var/www
+;chdir = /srv/http
 
 ; Redirect worker stdout and stderr into main error log. If not set, stdout and
 ; stderr will be redirected to /dev/null according to FastCGI specs.
