From 73611b24ed96ff131ce5a4b2f52d45755214b0b7 Mon Sep 17 00:00:00 2001
From: Jakub Kicinski <kubakici@wp.pl>
Date: Wed, 11 Dec 2013 18:12:05 +0100
Subject: [PATCH 02/12] Add coloring of diff patches inside emails.

---
 src/textview.c | 11 ++++++++++-
 src/textview.h |  1 +
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/src/textview.c b/src/textview.c
index 80c720035..fbe218c20 100644
--- a/src/textview.c
+++ b/src/textview.c
@@ -1077,6 +1077,7 @@ static void textview_write_body(TextView *textview, MimeInfo *mimeinfo)
 	
 	textview->is_in_signature = FALSE;
 	textview->is_diff = FALSE;
+	textview->is_in_git_patch = FALSE;
 	textview->is_attachment = FALSE;;
 
 	procmime_decode_content(mimeinfo);
@@ -1651,7 +1652,7 @@ static void textview_write_line(TextView *textview, const gchar *str,
 	}
 
 	if (prefs_common.enable_color) {
-		if (textview->is_diff) {
+		if (textview->is_diff || textview->is_in_git_patch) {
 			if (strncmp(buf, "+++ ", 4) == 0)
 				fg_color = "diff-add-file";
 			else if (buf[0] == '+')
@@ -1663,11 +1664,19 @@ static void textview_write_line(TextView *textview, const gchar *str,
 			else if (strncmp(buf, "@@ ", 3) == 0 &&
 				 strstr(&buf[3], " @@"))
 				fg_color = "diff-hunk";
+
+			if (strcmp(buf,"-- \n") == 0) {
+				textview->is_in_git_patch = FALSE;
+				textview->is_in_signature = TRUE;
+				fg_color = "signature";
+			}
 		} else if (account_signatures_matchlist_str_found(buf,"%s\n")
 				|| account_signatures_matchlist_str_found(buf, "- %s\n")
 				|| textview->is_in_signature) {
 			fg_color = "signature";
 			textview->is_in_signature = TRUE;
+		} else if (strncmp(buf, "diff --git ", 11) == 0) {
+			textview->is_in_git_patch = TRUE;
 		}
 	}
 
diff --git a/src/textview.h b/src/textview.h
index 35393a28b..104c1550a 100644
--- a/src/textview.h
+++ b/src/textview.h
@@ -66,6 +66,7 @@ struct _TextView
 	gboolean default_text;
 	gboolean is_in_signature;
 	gboolean is_diff;
+	gboolean is_in_git_patch;
 	gboolean is_attachment;
 	
 	GSList *uri_list;
-- 
2.15.1

