From f49e1f05139ea5316b1f121b45a9528a0588d1ce Mon Sep 17 00:00:00 2001
From: Jakub Kicinski <kubakici@wp.pl>
Date: Wed, 11 Dec 2013 18:12:05 +0100
Subject: [PATCH 02/10] Add coloring of diff patches inside emails.

---
 src/textview.c | 11 ++++++++++-
 src/textview.h |  1 +
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/src/textview.c b/src/textview.c
index ae93034f3..2ac617df0 100644
--- a/src/textview.c
+++ b/src/textview.c
@@ -1074,6 +1074,7 @@ static void textview_write_body(TextView *textview, MimeInfo *mimeinfo)
 	
 	textview->is_in_signature = FALSE;
 	textview->is_diff = FALSE;
+	textview->is_in_git_patch = FALSE;
 	textview->is_attachment = FALSE;;
 
 	procmime_decode_content(mimeinfo);
@@ -1629,7 +1630,7 @@ static void textview_write_line(TextView *textview, const gchar *str,
 	}
 
 	if (prefs_common.enable_color) {
-		if (textview->is_diff) {
+		if (textview->is_diff || textview->is_in_git_patch) {
 			if (strncmp(buf, "+++ ", 4) == 0)
 				fg_color = "diff-add-file";
 			else if (buf[0] == '+')
@@ -1641,11 +1642,19 @@ static void textview_write_line(TextView *textview, const gchar *str,
 			else if (strncmp(buf, "@@ ", 3) == 0 &&
 				 strstr(&buf[3], " @@"))
 				fg_color = "diff-hunk";
+
+			if (strcmp(buf,"-- \n") == 0) {
+				textview->is_in_git_patch = FALSE;
+				textview->is_in_signature = TRUE;
+				fg_color = "signature";
+			}
 		} else if (strcmp(buf,"-- \n") == 0
 				|| strcmp(buf, "- -- \n") == 0
 				|| textview->is_in_signature) {
 			fg_color = "signature";
 			textview->is_in_signature = TRUE;
+		} else if (strncmp(buf, "diff --git ", 11) == 0) {
+			textview->is_in_git_patch = TRUE;
 		}
 	}
 
diff --git a/src/textview.h b/src/textview.h
index 35393a28b..104c1550a 100644
--- a/src/textview.h
+++ b/src/textview.h
@@ -66,6 +66,7 @@ struct _TextView
 	gboolean default_text;
 	gboolean is_in_signature;
 	gboolean is_diff;
+	gboolean is_in_git_patch;
 	gboolean is_attachment;
 	
 	GSList *uri_list;
-- 
2.15.1

