From 1cb0c6d18d24d64ae38391f0b54696993e3b0d5c Mon Sep 17 00:00:00 2001
From: Jakub Kicinski <kubakici@wp.pl>
Date: Wed, 11 Dec 2013 18:12:05 +0100
Subject: [PATCH 02/11] Add coloring of diff patches inside emails.

---
 src/textview.c | 12 +++++++++++-
 src/textview.h |  2 ++
 2 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/src/textview.c b/src/textview.c
index 72f0efc90..ee6b2c8cd 100644
--- a/src/textview.c
+++ b/src/textview.c
@@ -1074,6 +1074,8 @@ static void textview_write_body(TextView *textview, MimeInfo *mimeinfo)
 	
 	textview->is_in_signature = FALSE;
 	textview->is_diff = FALSE;
+	textview->is_in_git_patch = FALSE;
+	textview->is_attachment = FALSE;;
 
 	procmime_decode_content(mimeinfo);
 
@@ -1620,7 +1622,7 @@ static void textview_write_line(TextView *textview, const gchar *str,
 	}
 
 	if (prefs_common.enable_color) {
-		if (textview->is_diff) {
+		if (textview->is_diff || textview->is_in_git_patch) {
 			if (strncmp(buf, "+++ ", 4) == 0)
 				fg_color = "diff-add-file";
 			else if (buf[0] == '+')
@@ -1632,11 +1634,19 @@ static void textview_write_line(TextView *textview, const gchar *str,
 			else if (strncmp(buf, "@@ ", 3) == 0 &&
 				 strstr(&buf[3], " @@"))
 				fg_color = "diff-hunk";
+
+			if (strcmp(buf,"-- \n") == 0) {
+				textview->is_in_git_patch = FALSE;
+				textview->is_in_signature = TRUE;
+				fg_color = "signature";
+			}
 		} else if (strcmp(buf,"-- \n") == 0
 				|| strcmp(buf, "- -- \n") == 0
 				|| textview->is_in_signature) {
 			fg_color = "signature";
 			textview->is_in_signature = TRUE;
+		} else if (strncmp(buf, "diff --git ", 11) == 0) {
+			textview->is_in_git_patch = TRUE;
 		}
 	}
 
diff --git a/src/textview.h b/src/textview.h
index 51b7643b4..104c1550a 100644
--- a/src/textview.h
+++ b/src/textview.h
@@ -66,6 +66,8 @@ struct _TextView
 	gboolean default_text;
 	gboolean is_in_signature;
 	gboolean is_diff;
+	gboolean is_in_git_patch;
+	gboolean is_attachment;
 	
 	GSList *uri_list;
 	gint body_pos;
-- 
2.14.1

