From 1128f68f1e4a0c442879b782bc9290f9e7215b5d Mon Sep 17 00:00:00 2001
From: Olivier Brunel <jjk@jjacky.com>
Date: Tue, 21 Feb 2017 18:52:06 +0100
Subject: [PATCH 8/8] Fix memory leak

Signed-off-by: Olivier Brunel <jjk@jjacky.com>
---
 src/folder.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/folder.c b/src/folder.c
index 458026ad7..97393d215 100644
--- a/src/folder.c
+++ b/src/folder.c
@@ -4806,6 +4806,8 @@ gint folder_item_search_msgs_local	(Folder			*folder,
 		}
 		processed_count++;
 
+		procmsg_msginfo_free(&msg);
+
 		if (progress_cb != NULL
 		    && !progress_cb(progress_data, FALSE, processed_count,
 			    matched_count, msgcount))
-- 
2.11.1

