From 3a14ca194836f25505f3728d009d8ca85deb6db2 Mon Sep 17 00:00:00 2001
From: jjacky <i.am.jack.mail@gmail.com>
Date: Tue, 19 Mar 2013 21:57:23 +0100
Subject: [PATCH] Remove open_basedir limitation


Signed-off-by: jjacky <i.am.jack.mail@gmail.com>
---
 include/sanity_check.php | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/include/sanity_check.php b/include/sanity_check.php
index 446a9ee..7dc3e5f 100644
--- a/include/sanity_check.php
+++ b/include/sanity_check.php
@@ -85,10 +85,6 @@
 				array_push($errors, "LOCK_DIRECTORY defined in config.php is not writable (chmod -R 777 ".LOCK_DIRECTORY.").\n");
 			}
 
-			if (ini_get("open_basedir")) {
-				array_push($errors, "PHP configuration option open_basedir is not supported. Please disable this in PHP settings file (php.ini).");
-			}
-
 			if (!function_exists("curl_init") && !ini_get("allow_url_fopen")) {
 				array_push($errors, "PHP configuration option allow_url_fopen is disabled, and CURL functions are not present. Either enable allow_url_fopen or install PHP extension for CURL.");
 			}
-- 
1.8.2

