--- ext/ieee.c.org	2010-11-10 05:41:02.000000000 +0100
+++ ext/ieee.c	2013-04-20 14:36:09.983516799 +0200
@@ -61,7 +61,8 @@
     rb_raise(rb_eTypeError, "wrong argument type %s (Float expected)",
 	     rb_class2name(CLASS_OF(vtmp)));
 #ifdef RUBY_1_9_LATER
-  gsl_ieee_fprintf_double(fp, &(RFLOAT_VALUE(vtmp)));
+  double _dbl = RFLOAT_VALUE(vtmp);
+  gsl_ieee_fprintf_double(fp, &_dbl);
 #else
   gsl_ieee_fprintf_double(fp, &(RFLOAT(vtmp)->value));
 #endif
