From f4fc2c382248ecce00bf3d7fb4db6cd0b9f577b4 Mon Sep 17 00:00:00 2001
From: Olivier Brunel <jjk@jjacky.com>
Date: Sun, 10 Jul 2016 21:07:08 +0200
Subject: [PATCH] getdtablesize: Fix returning hard instead of soft rlimit

Signed-off-by: Olivier Brunel <jjk@jjacky.com>
---
 src/legacy/getdtablesize.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/legacy/getdtablesize.c b/src/legacy/getdtablesize.c
index 682da6d..b30c193 100644
--- a/src/legacy/getdtablesize.c
+++ b/src/legacy/getdtablesize.c
@@ -7,5 +7,5 @@ int getdtablesize(void)
 {
 	struct rlimit rl;
 	getrlimit(RLIMIT_NOFILE, &rl);
-	return rl.rlim_max < INT_MAX ? rl.rlim_max : INT_MAX;
+	return rl.rlim_cur < INT_MAX ? rl.rlim_cur : INT_MAX;
 }
-- 
2.9.0

