From cdbb2cd130c302b087d5e9fef60146e09da618ee Mon Sep 17 00:00:00 2001
From: Olivier Brunel <jjk@jjacky.com>
Date: Wed, 31 Dec 2014 14:30:32 +0100
Subject: [PATCH] fix typo?

Signed-off-by: Olivier Brunel <jjk@jjacky.com>
---
 gtk/gtktooltip.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtk/gtktooltip.c b/gtk/gtktooltip.c
index fd2ba14..98242d0 100644
--- a/gtk/gtktooltip.c
+++ b/gtk/gtktooltip.c
@@ -1637,7 +1637,7 @@ _gtk_tooltip_handle_event (GdkEvent *event)
 	    hide_tooltip = !return_value;
 
 	    /* Leave notify should override the query function */
-	    hide_tooltip = (event->type == GDK_LEAVE_NOTIFY);
+	    hide_tooltip |= (event->type == GDK_LEAVE_NOTIFY);
 
 	    /* Is the pointer above another widget now? */
 	    if (GTK_TOOLTIP_VISIBLE (current_tooltip))
-- 
2.4.0

