From 4aade1ed42c4614cd566fe21b3f29f8d020659e9 Mon Sep 17 00:00:00 2001
From: Olivier Brunel <jjk@jjacky.com>
Date: Sat, 27 Sep 2014 13:57:38 +0200
Subject: [PATCH] treeview: Fix prelight re: gesture refactoring

Signed-off-by: Olivier Brunel <jjk@jjacky.com>
---
 gtk/gtktreeview.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gtk/gtktreeview.c b/gtk/gtktreeview.c
index 3eec8e4..1c42dae 100644
--- a/gtk/gtktreeview.c
+++ b/gtk/gtktreeview.c
@@ -4694,15 +4694,15 @@ gtk_tree_view_motion (GtkWidget      *widget,
 
   if (tree_view->priv->tree)
     {
+      new_y = MAX (0, TREE_WINDOW_Y_TO_RBTREE_Y (tree_view, event->y));
+
+      _gtk_rbtree_find_offset (tree_view->priv->tree, new_y, &tree, &node);
+
       /* If we are currently pressing down a button, we don't want to prelight anything else. */
       if (gtk_gesture_is_active (tree_view->priv->drag_gesture) ||
           gtk_gesture_is_active (tree_view->priv->multipress_gesture))
         node = NULL;
 
-      new_y = MAX (0, TREE_WINDOW_Y_TO_RBTREE_Y (tree_view, event->y));
-
-      _gtk_rbtree_find_offset (tree_view->priv->tree, new_y, &tree, &node);
-
       tree_view->priv->event_last_x = event->x;
       tree_view->priv->event_last_y = event->y;
       prelight_or_select (tree_view, tree, node, event->x, event->y);
-- 
2.6.1

