--- Zen-blua/gtk-2.0/xfce.rc	2013-08-10 13:23:29.000000000 +0200
+++ Zen-blua/gtk-2.0/xfce.rc.new	2013-08-12 10:41:12.628472117 +0200
@@ -9,9 +9,10 @@
 
 style "titlebar-adaptive" = "default"
 {
-	bg[SELECTED]		= shade(0.9, @selected_bg_color)
-	fg[SELECTED]        = mix(0.26, @bg_color, @selected_text_color)
-	fg[INSENSITIVE]		= mix(0.5, @bg_color, @fg_color)
+	bg[SELECTED]		= "#222c89"
+	fg[SELECTED]        = "#ffffff"
+	bg[INSENSITIVE]		= "#d4d4a6"
+	fg[INSENSITIVE]		= "#888a85"
 }
 
 style "titlebar-dark" = "default"
