--- iceweasel-41.0.1/mozilla/gfx/2d/Logging.h.orig	2015-10-11 23:18:21.614362889 +0200
+++ iceweasel-41.0.1/mozilla/gfx/2d/Logging.h	2015-10-11 21:12:33.117826547 +0200
@@ -11,9 +11,7 @@
 #include <stdio.h>
 #include <vector>
 
-#ifdef MOZ_LOGGING
 #include "mozilla/Logging.h"
-#endif
 
 #if defined(MOZ_WIDGET_GONK) || defined(MOZ_WIDGET_ANDROID)
 #include "nsDebug.h"
--- iceweasel-41.0.1/mozilla/dom/media/webm/WebMDemuxer.cpp~	2015-10-29 23:17:57.000000000 +0100
+++ iceweasel-41.0.1/mozilla/dom/media/webm/WebMDemuxer.cpp	2015-11-08 11:16:56.213634029 +0100
@@ -17,6 +17,7 @@
 #include "nsAutoRef.h"
 #include "NesteggPacketHolder.h"
 #include "XiphExtradata.h"
+#include "prprf.h"
 
 #include <algorithm>
 #include <stdint.h>
