--- test/api/Makefile.in.org	2013-06-07 21:16:48.873369541 +0200
+++ test/api/Makefile.in	2013-06-07 21:17:02.560035366 +0200
@@ -390,6 +390,7 @@
 @HAVE_GLIB_TRUE@test_unicode_CPPFLAGS = $(AM_CPPFLAGS) $(GLIB_CFLAGS) \
 @HAVE_GLIB_TRUE@	$(am__append_2)
 @HAVE_GLIB_TRUE@@HAVE_ICU_TRUE@test_unicode_LDADD = $(LDADD) $(top_builddir)/src/libharfbuzz-icu.la
+@HAVE_GLIB_TRUE@@HAVE_ICU_FALSE@test_unicode_LDADD = $(LDADD) $(top_builddir)/src/libharfbuzz.la
 @HAVE_GLIB_TRUE@test_cplusplus_SOURCES = test-cplusplus.cc
 @HAVE_GLIB_TRUE@test_c_CPPFLAGS = $(AM_CPPFLAGS) $(am__append_4) \
 @HAVE_GLIB_TRUE@	$(am__append_6)
